### PR TITLE
Update baseline vs. NETFX

### PR DESCRIPTION
**Pull request by danmosemsft**
First commit runs `msbuild /p:baselineapicompat=true ApiCompat.proj` in `src\shims` to catch up with recent changes. 

Second commit moves some of these to ignore file that should be ignored and moves back a bunch of entries for missing types from the ignore file that didn't have an obvious reason for being ignored.

### CODE REVIEW

File: src/shims/ApiCompatBaseline.netcoreapp.netfx461.ignore.txt [hunk 1]
```diff
@@ -1059,410 +1061,12 @@ TypesMustExist : Type 'System.Runtime.Remoting.Proxies.RealProxy' does not exist
...
-TypesMustExist : Type 'System.Security.AccessControl.CryptoKeyRights' does not exist in the implementation but it does exist in the contract.
-TypesMustExist : Type 'System.Security.AccessControl.CryptoKeySecurity' does not exist in the implementation but it does exist in the contract.
-TypesMustExist : Type 'System.Security.Cryptography.CryptoAPITransform' does not exist in the implementation but it does exist in the contract.
-TypesMustExist : Type 'System.Security.Cryptography.HMACRIPEMD160' does not exist in the implementation but it does exist in the contract.
-TypesMustExist : Type 'System.Security.Cryptography.MACTripleDES' does not exist in the implementation but it does exist in the contract.
-TypesMustExist : Type 'System.Security.Cryptography.RijndaelManagedTransform' does not exist in the implementation but it does exist in the contract.
-TypesMustExist : Type 'System.Security.Cryptography.RIPEMD160' does not exist in the implementation but it does exist in the contract.
-TypesMustExist : Type 'System.Security.Cryptography.RIPEMD160Managed' does not exist in the implementation but it does exist in the contract.
```
Comment (bartonjs) [inline src/shims/ApiCompatBaseline.netcoreapp.netfx461.ignore.txt]: CryptoAPITransform through RIPEMD160Managed are types that we consciously are not bringing to core.
CryptoKeyAccessRule through CryptoKeySecurity are part of the complications of Windows ACLs on a cross-platform system.

Comment (danmosemsft) [reply src/shims/ApiCompatBaseline.netcoreapp.netfx461.ignore.txt]: Thanks. I've paged out of my memory many of these discussions..

Comment (danmosemsft) [reply src/shims/ApiCompatBaseline.netcoreapp.netfx461.ignore.txt]: fixed

**Comment by danmosemsft** [comment]
@weshaggard do you recall context on any of the additions to the non-ignore file? If so I can move to ignore with a note.

File: src/shims/ApiCompatBaseline.netcoreapp.netfx461.txt [hunk 2]
```diff
@@ -2,56 +2,254 @@ ApiCompat Error: 0 : Unable to resolve assembly 'Assembly(Name=System.ServiceMod
...
-ApiCompat Error: 0 : Unable to resolve assembly 'Assembly(Name=System.IO.FileSystem.AccessControl, Version=4.0.3.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a)' referenced by the contract assembly 'Assembly(Name=System.DirectoryServices, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a)'.
-ApiCompat Error: 0 : Unable to resolve assembly 'Assembly(Name=System.Security.AccessControl, Version=4.1.1.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a)' referenced by the contract assembly 'Assembly(Name=System.DirectoryServices, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a)'.
-ApiCompat Error: 0 : Unable to resolve assembly 'Assembly(Name=netstandard, Version=2.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51)' referenced by the contract assembly 'Assembly(Name=System.DirectoryServices.Protocols, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a)'.
+Compat issues with assembly mscorlib:
+TypeCannotChangeClassification : Type 'System.RuntimeArgumentHandle' is a 'ref struct' in the implementation but is a 'struct' in the contract.
```
Comment (weshaggard) [inline src/shims/ApiCompatBaseline.netcoreapp.netfx461.txt]: These 'ref struct' issues can be added to the ignore list as we don't plan to update them in .NET Framework.

File: src/shims/ApiCompatBaseline.netcoreapp.netfx461.txt [hunk 3]
```diff
@@ -2,56 +2,254 @@ ApiCompat Error: 0 : Unable to resolve assembly 'Assembly(Name=System.ServiceMod
...
+TypesMustExist : Type 'System.Runtime.ProfileOptimization' does not exist in the implementation but it does exist in the contract.
+TypesMustExist : Type 'System.Runtime.Serialization.Formatters.InternalRM' does not exist in the implementation but it does exist in the contract.
+TypesMustExist : Type 'System.Runtime.Serialization.Formatters.InternalST' does not exist in the implementation but it does exist in the contract.
+TypesMustExist : Type 'System.Runtime.Serialization.Formatters.ISoapMessage' does not exist in the implementation but it does exist in the contract.
+TypesMustExist : Type 'System.Runtime.Serialization.Formatters.ServerFault' does not exist in the implementation but it does exist in the contract.
+TypesMustExist : Type 'System.Runtime.Serialization.Formatters.SoapFault' does not exist in the implementation but it does exist in the contract.
+TypesMustExist : Type 'System.Runtime.Serialization.Formatters.SoapMessage' does not exist in the implementation but it does exist in the contract.
+TypesMustExist : Type 'System.Security.Policy.ApplicationSecurityInfo' does not exist in the implementation but it does exist in the contract.
```
Comment (weshaggard) [inline src/shims/ApiCompatBaseline.netcoreapp.netfx461.txt]: A lot of these are tied to remoting/CAS so if we don't plan to support that then we can keep these excluded.

File: src/shims/ApiCompatBaseline.netcoreapp.netfx461.txt [hunk 4]
```diff
@@ -2,56 +2,254 @@ ApiCompat Error: 0 : Unable to resolve assembly 'Assembly(Name=System.ServiceMod
...
+TypesMustExist : Type 'System.Security.Authentication.ExtendedProtection.Configuration.ServiceNameElement' does not exist in the implementation but it does exist in the contract.
+TypesMustExist : Type 'System.Security.Authentication.ExtendedProtection.Configuration.ServiceNameElementCollection' does not exist in the implementation but it does exist in the contract.
+TypesMustExist : Type 'System.Security.Claims.DynamicRoleClaimProvider' does not exist in the implementation but it does exist in the contract.
+TypesMustExist : Type 'System.Web.AspNetHostingPermission' does not exist in the implementation but it does exist in the contract.
+TypesMustExist : Type 'System.Web.AspNetHostingPermissionAttribute' does not exist in the implementation but it does exist in the contract.
+TypesMustExist : Type 'System.Web.AspNetHostingPermissionLevel' does not exist in the implementation but it does exist in the contract.
+CannotRemoveBaseTypeOrInterface : Type 'System.Windows.Markup.ValueSerializerAttribute' does not implement interface 'System.Runtime.InteropServices._Attribute' in the implementation but it does in the contract.
```
Comment (weshaggard) [inline src/shims/ApiCompatBaseline.netcoreapp.netfx461.txt]: Anything with the "_" as the type name were for COM stuff which we aren't supporting.

**Comment by weshaggard** [comment]
I had a quick scan and left a few notes for things I remember off the top of my head.

**Pull request merged**